lib/matplotlib/testing/marks.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import shutil
+
+import pytest
+
+import matplotlib.testing
+from matplotlib import _get_executable_info, ExecutableNotFoundError
+
+
+_log = logging.getLogger(__name__)
+
+
+def _checkdep_usetex():
+    if not shutil.which("tex"):
+        _log.warning("usetex mode requires TeX.")
+        return False
+    try:
+        _get_executable_info("dvipng")
+    except ExecutableNotFoundError:
+        _log.warning("usetex mode requires dvipng.")
+        return False
+    try:
+        _get_executable_info("gs")
+    except ExecutableNotFoundError:
+        _log.warning("usetex mode requires ghostscript.")
+        return False
+    return True
+
+
+needs_ghostscript = pytest.mark.skipif(
```

Comment:
By putting them here, we're technically making them public: `matplotlib.testing.marks.needs_ghostscript`.

While I'm not sure whether `testing` itself should be public. Let's not add to it. Either `_marks.py` or prefix all the markers. I'd go with the first, similar to `_api`.

Oh, and pytest calls these ["markers"](https://docs.pytest.org/en/6.2.x/example/markers.html?highlight=markers). so, `_markers.py` it is.

Reply:
Not critical here, but I still don't think these are technically markers because they aren't in the `pytest.mark` namespace, which means someone can't skip these from the command line with something like: `pytest -m "not latex"`.

Reply:
I was also thinking about that after reading up a bit further. On the other hand, I do not think you can do `pytest -m "not skipif"` either? So marker seems to have a rather wide interpretation.

Reply:
Turns out you can actually do that, `pytest -m skipif`.

Should I rename it to `_decorators.py` then?

Reply:
I guess one can add them as proper markers, but I had a quick look and it seemed to require quite a bit more effort than possibly worth it. As far as I can tell one would need to create a proper `conftest.py` with functions checking the markers and so on, not simply "inserting" a named marker derived from another marker. But I could have been bad at searching.

(Which is basically what you stated in an earlier comment.)